Source/WTF/wtf/RunLoop.h

```diff
@@ -114,11 +115,62 @@ class WTF_CAPABILITY("is current") RunLoop final : public GuaranteedSerialFuncti
 
     WTF_EXPORT_PRIVATE void threadWillExit();
 
+    enum class Activity : uint8_t {
+        BeforeWaiting   = 1 << 0,
+        Entry           = 1 << 1,
+        Exit            = 1 << 2,
+        AfterWaiting    = 1 << 3
+    };
+
 #if USE(GLIB_EVENT_LOOP)
     WTF_EXPORT_PRIVATE GMainContext* mainContext() const { return m_mainContext.get(); }
-    enum class Event { WillDispatch, DidDispatch };
-    using Observer = WTF::Observer<void(Event, const String&)>;
-    WTF_EXPORT_PRIVATE void observe(const Observer&);
+
+    // Event observers (only used for PageTimelineAgent)
+    enum class Event : bool {
+        WillDispatch,
+        DidDispatch
+    };
+
+    using EventObserver = Observer<void(Event, const String&)>;
+    WTF_EXPORT_PRIVATE void observeEvent(const EventObserver&);
+
+    // Activity observers (used to implement WebCore::RunLoopObserver)
+    class ActivityObserver {
```

Comment:
Can this be moved to its own file? I think it's too much glib specific code for a cross-platform header.

Reply:
Moving to glib/ActivityObserver.h.